src/inspector_agent.h

```diff
@@ -121,7 +123,13 @@ class Agent {
   std::unique_ptr<InspectorIo> io_;
   std::unique_ptr<ParentInspectorHandle> parent_handle_;
   std::string path_;
-  std::shared_ptr<DebugOptions> debug_options_;
+
+  // This is a copy of the debug options parsed from CI in the Environment.
+  // Do not use the host_port in that, instead manipulate the shared host_port_
+  // pointer which is meant to store the actual host and port of the inspector
+  // server.
```

Comment:
I've tried a few approaches, e.g. splitting the options into two `Options` subclasses instead of putting them in one `DebugOptions`, that makes the code even more complicated because we have combined arguments like `--inspect-brk=127.0.0.1`. Also, the DebugOptions are actually per-process (because those are parsed from CLI) whereas the shared HostPort are per-Environment. Conceptually, they are not really duplicates because the CLI ones are "snapshots" of the parsed arguments which may still be useful themselves (e.g. in tests)